components/datagrid/theme.css

```diff
@@ -93,6 +93,13 @@
   color: var(--color-teal-darkest);
 }
 
+.loading-bar {
+  position: absolute;
+  top: 48px;
+  left: 0;
+  z-index: 3;
```

Comment:
FYI: This is the same value as the HeaderRowOverlay (this is where the bulk actions etc. are displayed)